.github/workflows/inspektor-gadget.yml

```diff
@@ -1473,6 +1473,84 @@ jobs:
         gadget_tag: ${{ steps.set-repo-determine-image-tag.outputs.gadget-tag }}
         test_summary_suffix: "aro"
 
+  test-integration-eks:
+    name: Integration tests on EKS
+    # level: 4
+    needs:
+      - test-unit
+      - build-clients
+      - build-ig
+      - build-gadget-container-images
+      - publish-gadget-images-manifest
+      - build-and-push-gadgets
+    runs-on: ubuntu-latest
+    # These permissions are needed to interact with GitHub's OIDC Token endpoint.
+    permissions:
+      id-token: write
+      contents: read
+    strategy:
+      fail-fast: false
+      matrix:
+        arch: [amd64, arm64]
+    env:
+      AWS_REGION: us-east-2
+      AWS_EKS_CLUSTER_PREFIX: ig-ci-eks-
+    steps:
+    - uses: actions/checkout@v4
+    - name: Setup go
+      uses: actions/setup-go@v5
+      with:
+        go-version: ${{ env.GO_VERSION }}
+        cache: true
+    - name: Set container repository and determine image tag
+      id: set-repo-determine-image-tag
+      uses: ./.github/actions/set-container-repo-and-determine-image-tag
+      with:
+        registry: ${{ env.REGISTRY }}
+        container-image: ${{ env.CONTAINER_REPO }}
+    - name: Craft cluster name
+      shell: bash
+      run: |
+        echo "CLUSTER_NAME=${{ env.AWS_EKS_CLUSTER_PREFIX }}${{ matrix.arch }}-${RANDOM}" >> $GITHUB_ENV
+    - name: Install eksctl
+      run: |
+        ARCH=amd64
+        PLATFORM=$(uname -s)_$ARCH
+        curl -sLO "https://github.com/eksctl-io/eksctl/releases/latest/download/eksctl_$PLATFORM.tar.gz"
+        curl -sL "https://github.com/eksctl-io/eksctl/releases/latest/download/eksctl_checksums.txt" | grep $PLATFORM | sha256sum --check
+        tar -xzf eksctl_$PLATFORM.tar.gz -C /tmp && rm eksctl_$PLATFORM.tar.gz
+        sudo mv /tmp/eksctl /usr/local/bin
+    - name: Configure AWS credentials
+      uses: aws-actions/configure-aws-credentials@v4
+      with:
+        role-to-assume: arn:aws:iam::637423639163:role/gha-role
```

Comment:
I got a little scare after seeing the account id (`637423639163`) here but reconfirmed that it [is NOT a sensitive information](https://www.lastweekinaws.com/blog/are-aws-account-ids-sensitive-information/) so all good.

Reply:
Even if it is not a sensitive information, that would be better to use a GitHub secret, so that forks of the IG repository can choose to enable this CI with their own accounts (or alternatively, disable this CI when the environment variable is not set).

Reply:
Rather than a secret I used a [repository variable,](https://docs.github.com/en/actions/learn-github-actions/variables#creating-configuration-variables-for-a-repository) so other folks are able to enable and configure it for their accounts if they want to.